.github/workflows/common/openapi-build/action.yaml

```diff
@@ -0,0 +1,43 @@
+name: Build and Publish OpenAPI UI
+description: "Build and publish OpenAPI UI for a service"
+
+inputs:
+  service:
+    description: "The name of the service to build OpenAPI UI for"
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Install Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: 20
+
+    - name: Install dependencies
+      run: npm ci
+      shell: bash
+      working-directory: ./services/${{ inputs.service }}
+
+    - name: Generate Swagger Metadata
+      run: npm run build:metadata
+      shell: bash
+      working-directory: ./services/${{ inputs.service }}
+
+    - name: Generate OpenAPI/Swagger JSON
+      run: npm run build:swagger
+      shell: bash
+      working-directory: ./services/${{ inputs.service }}
+
+    - name: Generate OpenAPI/Swagger UI
+      run: npm run generate-swagger-ui
+      shell: bash
+      working-directory: ./services/${{ inputs.service }}
```

Comment:
FYI, I'm about to publish a PR standardizing on the following pattern for these targets:
```
generate:metadata
generate:openapi
generate:swagger-ui
```

Once this is merged, I'll pull into my PR & update the workflows.